homeassistant/components/hive/config_flow.py

```diff
@@ -90,24 +92,48 @@ async def async_step_2fa(self, user_input=None):
 
             if not errors:
                 try:
+                    if self.context["source"] == config_entries.SOURCE_REAUTH:
+                        return await self.async_setup_hive_entry()
                     self.device_registration = True
-                    return await self.async_setup_hive_entry()
+                    return await self.async_step_configuration()
                 except UnknownHiveError:
                     errors["base"] = "unknown"
 
         schema = vol.Schema({vol.Required(CONF_CODE): str})
         return self.async_show_form(step_id="2fa", data_schema=schema, errors=errors)
 
+    async def async_step_configuration(self, user_input=None):
+        """Handle hive configuration step."""
+        errors = {}
+
+        if user_input:
+            if not errors:
+                try:
+                    if self.device_registration:
+                        await self.hive_auth.device_registration(
+                            user_input["device_name"]
+                        )
+                        self.data[
+                            "device_data"
+                        ] = await self.hive_auth.get_device_data()
+                    return await self.async_setup_hive_entry()
+                except UnknownHiveError:
+                    errors["base"] = "unknown"
+
+        schema = vol.Schema(
+            {vol.Optional(CONF_DEVICE_NAME, default="Home Assistant"): str}
```

Comment:
In the case of an error won't you want to redisplay the form with the form pre-filled with what the user typed?

Reply:
That's done now